src/consensus/vote.cpp

```diff
@@ -213,44 +208,66 @@ uint64_t VoteManager::getVerifiedVotesSize() const {
   uint64_t size = 0;
 
   sharedLock_ lock(verified_votes_access_);
-  std::map<uint64_t, std::unordered_map<vote_hash_t, Vote>>::const_iterator it;
-  for (it = verified_votes_.begin(); it != verified_votes_.end(); ++it) {
-    size += it->second.size();
+  for (auto const& round : verified_votes_) {
+    for (auto const& step : round.second) {
+      for (auto const& voted_value : step.second) {
+        size += voted_value.second.size();
+      }
+    }
   }
 
   return size;
 }
 
 void VoteManager::addVerifiedVote(Vote const& vote) {
-  auto pbft_round = vote.getRound();
+  auto round = vote.getRound();
+  auto step = vote.getStep();
+  auto voted_value = vote.getBlockHash();
   auto hash = vote.getHash();
 
-  {
-    upgradableLock_ lock(verified_votes_access_);
-    std::map<uint64_t, std::unordered_map<vote_hash_t, Vote>>::const_iterator found_round =
-        verified_votes_.find(pbft_round);
-    if (found_round != verified_votes_.end()) {
-      std::unordered_map<vote_hash_t, Vote>::const_iterator found_vote = found_round->second.find(hash);
-      if (found_vote != found_round->second.end()) {
-        LOG(log_dg_) << "Vote " << hash << " is in verified map already";
-        return;
+  upgradableLock_ lock(verified_votes_access_);
+  if (verified_votes_.count(round)) {
```

Comment:
how about using if statement with initializer and find instead of count:
`if (auto verified_votes_it = verified_votes_.find(round); verified_votes_it != verified_votes_.end())`

This way you dont need to always lookup roung by `verified_votes_[round]`, but you will use  verified_votes_it. Same logic applies for step, voted_value....

Reply:
Fixed, thanks